File: lib/src/main/java/com/darklycoder/wifitool/lib/info/action/IWiFiAction.java
package com.darklycoder.wifitool.lib.info.action;

import android.support.annotation.NonNull;

import com.darklycoder.wifitool.lib.type.Types;

public abstract class IWiFiAction {

    @Types.ActionStateType
    private int actionState;

    IWiFiAction() {
        setState(Types.ActionStateType.WAITING);
    }

    private String getActionName() {
        return getClass().getSimpleName();
    }

    public int getActionState() {
        return actionState;
    }

    public void setState(@Types.ActionStateType int state) {
        this.actionState = state;
    }

    public void end() {
        this.actionState = Types.ActionStateType.END;
    }

    @NonNull
    @Override
    public String toString() {
        return getActionName() + " | actionState:" + actionState;
    }

}

File: lib/src/main/java/com/darklycoder/wifitool/lib/info/action/IWiFiAction.kt
package com.darklycoder.wifitool.lib.info.action

import com.darklycoder.wifitool.lib.type.Types

abstract class IWiFiAction internal constructor() {

    @Types.ActionStateType
    var actionState: Int = 0
        private set

    private val actionName: String
        get() = javaClass.simpleName

    init {
        setState(Types.ActionStateType.WAITING)
    }

    fun setState(@Types.ActionStateType state: Int) {
        this.actionState = state
    }

    open fun end() {
        this.actionState = Types.ActionStateType.END
    }

    override fun toString(): String {
        return "$actionName | actionState:$actionState"
    }

}
